hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/supervisor/workload/SubscriptionAssignmentView.java

```diff
@@ -0,0 +1,204 @@
+package pl.allegro.tech.hermes.consumers.supervisor.workload;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Sets;
+import pl.allegro.tech.hermes.api.SubscriptionName;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.BiConsumer;
+
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toSet;
+
+public class SubscriptionAssignmentView {
```

Comment:
I don't quite feel this `SubscriptionAssignmentView` name. Usually when I think of a view I mean a reflection of some underlying state that gets updated when the state changes. This object however does represent the state itself, it represents a composition of multiple subscription assignments. How about `WorkAssignment` instead?

Feel free to argue though, you're the domain expert here ;)